doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -15,46 +12,63 @@ feature, and submitting pull requests.
 
 .. note:: A `contributor license agreement`_ must be signed before any contributing pull requests can be merged.
 
-Developer Workflow
-------------------
-
 Overview:
 
-#. `Discuss design changes or big development efforts`_
-#. `Get set up`_
-#. `Create new branch`_
-#. `Develop and test contributions locally`_
+#. `Choosing a task`_
+#. `Development`_
+
+    #. `Get set up`_
+    #. `Create new branch`_
+    #. `Develop and test contributions locally`_
+
+       #. `Add new tests`_
+
+#. `Design`_
```

Comment:
Should the Design section come before the Development section since (in many cases) design should occur before development?  It could have a short-circuit to the development section if no design is required (e.g., the first subsection could be "Does the effort require design?" where some criteria would be whether it's a change to a user facing feature or interface, whether it's a change to the software architecture, etc.  The text could say something about "in many cases it's reasonable to start development without a full design review, but such efforts should be open to changing once the design is being discussed."

Reply:
We were debating that and tried to get a similar effect in this section:

https://github.com/chapel-lang/chapel/pull/14957/files#diff-931c81291714b774c2ad29f6e43f6cf7R56-R63

Reply:
Do you think it would be better to short-circuit the design section rather than have a link to it from the first section?

Our thought process was that for most first-time contributors, they will be picking up a reasonably well-defined task, and will just need to comment on that issue. Therefore, the [Development] section is most likely what they will need when getting started.

Reply:
I think we've had first-time contributors jump into tasks that have a design component unwittingly (and then get bogged down / discouraged by the fact that they have to take others' opinions into account), which is what made me suggest putting it first but having a clause defining what should trigger design discussions and encouraging people to skip right to development if it doesn't apply.  But I'm OK with whatever you guys decide.

Reply:
I just remembered that the previous version of this document had [Discuss design changes or big development efforts] as the first step, and this step was converted to [Choosing a task] with a link to the more detailed [Design] section if a proper design discussion is warranted.

I am leaning towards leaving the structure as is, but we can wait to see what others think as well. (@mppf?)

Reply:
I like Brad's approach (put Design first, but put a sentence in the beginning of it about when design is not needed).

Reply:
Our previous GOTO [design] statement followed this pattern:

1. If an issue exists, comment on it
       - If that's a big or complicated task, see [design] section for guidance on how to proceed
2. If an issue does not exist, create one and see [design] section for guidance

Trying to marry that to this pattern:

Design discussion is needed when changes impact:

- language or standard library
- software architecture

Any thoughts? Maybe it would be easier to list situations in which design discussion is _not_ required...

Reply:
See latest commit for my draft of this.